src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/MemberExpansion.cs

```diff
@@ -437,24 +445,31 @@ internal StaticMembersExpansion(Type declaredType, Expansion members)
             MemberAndDeclarationInfo member,
             DkmClrValue memberValue,
             EvalResultDataItem parent,
+            DynamicFlagsMap dynamicFlagsMap,
             ExpansionFlags flags)
         {
-            var formatter = resultProvider.Formatter;
+            var declaredType = member.Type;
+            var declaredTypeInfo = dynamicFlagsMap.SubstituteDynamicFlags(member.OriginalDefinitionType, new DynamicFlagsCustomTypeInfo(member.TypeInfo)).GetCustomTypeInfo();
             string memberName;
+            // Considering, we're not handling the case of a member inherited from a generic base type.
             var typeDeclaringMember = member.GetExplicitlyImplementedInterface(out memberName) ?? member.DeclaringType;
+            var typeDeclaringMemberInfo = typeDeclaringMember.IsInterface
+                ? dynamicFlagsMap.SubstituteDynamicFlags(typeDeclaringMember.GetInterfaceListEntry(member.DeclaringType), originalDynamicFlags: default(DynamicFlagsCustomTypeInfo)).GetCustomTypeInfo()
+                : null;
+            var formatter = resultProvider.Formatter;
             memberName = formatter.GetIdentifierEscapingPotentialKeywords(memberName);
             var fullName = MakeFullName(
                 formatter,
                 memberName,
-                typeDeclaringMember,
+                new TypeAndCustomInfo(typeDeclaringMember, typeDeclaringMemberInfo), // Note: Won't include DynamicAttribute.
                 member.RequiresExplicitCast,
                 member.IsStatic,
                 parent);
             return resultProvider.CreateDataItem(
                 inspectionContext,
                 memberName,
-                typeDeclaringMember: (member.IncludeTypeInMemberName || typeDeclaringMember.IsInterface) ? typeDeclaringMember : null,
-                declaredType: member.Type,
+                typeDeclaringMemberAndInfo: new TypeAndCustomInfo((member.IncludeTypeInMemberName || typeDeclaringMember.IsInterface) ? typeDeclaringMember : null, typeDeclaringMemberInfo), // Note: Won't include DynamicAttribute.
```

Comment:
Might this be a TypeAndCustomInfo with null type and non-empty info? Is that valid?